graph/path/bellman_ford_moore.go

```diff
@@ -27,46 +30,100 @@ func BellmanFordFrom(u graph.Node, g graph.Graph) (path Shortest, ok bool) {
 	path = newShortestFrom(u, nodes)
 	path.dist[path.indexOf[u.ID()]] = 0
 
+	// Queue to keep track which nodes need to be relaxed.
+	// Only nodes whose vertex distance changed in the previous iterations need to be relaxed again.
+	queue := newBellmanFordQueue(path.indexOf)
+	queue.enqueue(u)
+
+	n := len(nodes)
+	// The maximum of edges in a graph is |V| * (|V| -1) which is also the worst case complexity.
+	// If the queue-loop has more iterations than the amount of maximum edges
+	// it indicates that we have a negative cycle.
+	maxEdges := n * (n - 1)
+	negativeCycle := false
+
+	loops := 0
```

Comment:
`var loops int` and place this immediately after `maxEdges := len(nodes) * (len(nodes) - 1)`.